edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test.Common/TempDirectory.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Util.Test.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+
+    public class TempDirectory : IDisposable
+    {
+        private List<string> dirs = new List<string>();
+
+        protected string GetTempDir()
```

Comment:
I think temp dir created in this method should be owned by caller.  therefore you don't need to implement IDisposable for this class.

Reply:
Implementing IDisposable is the point of this class. Since xunit calls Dispose on the test classes, all test classes that want a temp directory can inherit this and not worry about cleanup. Alternatively, it can be wrapped in a using block.